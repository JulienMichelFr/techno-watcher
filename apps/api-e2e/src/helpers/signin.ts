import { AuthResponseModel } from '@techno-watcher/api-models';
import * as supertest from 'supertest';

export async function signIn(email: string = 'user1@email.com', password: string = 'Pas$w0rd'): Promise<AuthResponseModel> {
  const { body } = await supertest('http://localhost:3333').post('/auth/sign-in').send({ email, password }).expect(200);
  return body;
}
