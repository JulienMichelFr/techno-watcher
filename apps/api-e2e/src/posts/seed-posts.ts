import { Comment, Invitation, Post, PrismaClient, User } from '@prisma/client';
import { rand, randBetweenDate, randCatchPhrase, randNumber, randParagraph, randUrl, randWord, seed as falsoSeed } from '@ngneat/falso';
import * as bcrypt from 'bcrypt';

const prisma: PrismaClient = new PrismaClient();

falsoSeed('seed1');

//#region Seed utils
function generateCount<T>(count: number, fn: (index: number) => T): T[] {
  return Array.from({ length: count }).map((_, index) => fn(index));
}

function hashPassword(password: string): string {
  return bcrypt.hashSync(password, 12);
}

function getDate(from: Date = new Date('01/01/2022'), to: Date = new Date('12/31/2022')): Date {
  return randBetweenDate({ from, to });
}

//#endregion

async function seedInvitations(): Promise<Invitation[]> {
  let invitationId: number = 0;

  function createInvitation(): { id: number; code: string } {
    return {
      id: ++invitationId,
      code: randWord(),
    };
  }

  await prisma.invitation.deleteMany({});
  await prisma.invitation.createMany({
    data: generateCount(2, createInvitation),
  });
  return prisma.invitation.findMany({
    orderBy: {
      id: 'asc',
    },
  });
}

async function seedUsers(invitations: Invitation[]): Promise<User[]> {
  await prisma.user.createMany({
    data: [
      {
        id: 1,
        email: 'user1@email.com',
        password: hashPassword('Pas$w0rd'),
        username: 'User 1',
        invitationId: invitations[0].id,
        createdAt: new Date(2022, 0, 1),
        updatedAt: new Date(2022, 0, 1),
      },
      {
        id: 2,
        email: 'user2@email.com',
        password: hashPassword('Pas$w0rd'),
        username: 'User 2',
        invitationId: invitations[1].id,
        createdAt: new Date(2022, 0, 1),
        updatedAt: new Date(2022, 0, 1),
      },
    ],
  });

  return prisma.user.findMany({
    orderBy: {
      id: 'asc',
    },
  });
}

async function seedPosts(users: User[]): Promise<Post[]> {
  let postId: number = 0;

  function createPost(): Post {
    const tags: string[] = generateCount(randNumber({ min: 0, max: 3 }), () => randWord());
    const date: Date = getDate();

    return {
      id: ++postId,
      link: randUrl(),
      title: randCatchPhrase(),
      content: randParagraph(),
      authorId: rand(users).id,
      tags,
      createdAt: date,
      updatedAt: date,
      deletedAt: null,
    };
  }

  await prisma.post.createMany({
    data: generateCount(15, createPost),
  });

  return prisma.post.findMany({
    orderBy: {
      id: 'asc',
    },
  });
}

async function seedComments(users: User[], posts: Post[]): Promise<Comment[]> {
  let commentId: number = 0;
  const postCommentMap: Map<number, number[]> = new Map<number, number[]>();
  posts.forEach(({ id }) => postCommentMap.set(id, []));

  function createComment(): Comment {
    const currentCommentId: number = ++commentId;
    const date: Date = getDate();
    const postId: number = rand(posts).id;
    let parentCommentId: number | null = null;
    const commentIds: number[] = postCommentMap.get(postId);
    if (commentIds?.length) {
      parentCommentId = rand(commentIds);
    }
    commentIds.push(currentCommentId);

    return {
      id: currentCommentId,
      content: randParagraph(),
      authorId: rand(users).id,
      postId,
      createdAt: date,
      updatedAt: date,
      deletedAt: null,
      parentCommentId,
    };
  }

  await prisma.comment.createMany({
    data: generateCount(25, createComment),
  });

  return prisma.comment.findMany({
    orderBy: {
      id: 'asc',
    },
  });
}

export async function clearDatabase(): Promise<void> {
  await Promise.all([prisma.comment.deleteMany({}), prisma.post.deleteMany({}), prisma.user.deleteMany({}), prisma.invitation.deleteMany({})]);
}

export async function seed(): Promise<void> {
  await clearDatabase();

  const invitations: Invitation[] = await seedInvitations();
  const users: User[] = await seedUsers(invitations);
  const posts: Post[] = await seedPosts(users);
  const comments: Comment[] = await seedComments(users, posts);

  await prisma.$disconnect();
}
