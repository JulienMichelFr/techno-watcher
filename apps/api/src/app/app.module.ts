import { Module } from '@nestjs/common';

import { PostModule } from './modules/post/post.module';
import { UserModule } from './modules/user/user.module';
import { AuthModule } from './modules/auth/auth.module';
import { PrismaModule } from './modules/prisma/prisma.module';
import { ConfigModule } from '@nestjs/config';

@Module({
  imports: [ConfigModule.forRoot({ isGlobal: true }), PostModule, UserModule, AuthModule, PrismaModule],
  controllers: [],
  providers: [],
})
export class AppModule {}
