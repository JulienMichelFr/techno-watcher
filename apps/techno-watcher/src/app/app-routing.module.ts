import { NgModule } from '@angular/core';
import { Route, RouterModule } from '@angular/router';
import { HomePageComponent } from './pages/home-page/home-page.component';

const Routes: Route[] = [{ path: '', component: HomePageComponent }];

@NgModule({
  imports: [RouterModule.forRoot(Routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
