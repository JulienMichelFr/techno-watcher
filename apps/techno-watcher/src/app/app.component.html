<techno-watcher-navbar></techno-watcher-navbar>
<div class="router-container">
  <router-outlet></router-outlet>
</div>
