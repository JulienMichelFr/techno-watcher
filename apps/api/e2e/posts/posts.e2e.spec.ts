import * as supertest from 'supertest';
import { signIn } from '../helpers/signin';
import { seed } from './seed-posts';
import { INestApplication, ValidationPipe } from '@nestjs/common';
import { Test, TestingModule } from '@nestjs/testing';
import { AppModule } from '../../src/app/app.module';

describe('Posts', () => {
  let bearerToken: string;
  let app: INestApplication;

  beforeAll(async () => {
    await seed();

    const moduleRef: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleRef.createNestApplication();
    app.enableCors();
    app.useGlobalPipes(
      new ValidationPipe({
        whitelist: true,
        transform: true,
      })
    );
    await app.init();
    const { accessToken } = await signIn(app);
    bearerToken = `Bearer ${accessToken}`;
  });

  it('GET /posts', async () => {
    return supertest(app.getHttpServer())
      .get('/posts')
      .expect(200)
      .expect((response) => {
        expect(response.body).toMatchSnapshot();
      });
  });

  it('GET /posts?skip=10&take=10', async () => {
    return supertest(app.getHttpServer())
      .get('/posts?skip=10&take=10')
      .expect(200)
      .expect((response) => {
        expect(response.body).toMatchSnapshot();
      });
  });

  it('GET /posts?sort=id:ASC', async () => {
    return supertest(app.getHttpServer())
      .get('/posts?sort=id:asc')
      .expect(200)
      .expect((response) => {
        expect(response.body).toMatchSnapshot();
      });
  });

  it('GET /posts/1013', async () => {
    return supertest(app.getHttpServer())
      .get('/posts/1013')
      .expect(200)
      .expect((response) => {
        expect(response.body).toMatchSnapshot();
      });
  });

  it('POST /posts', async () => {
    return supertest(app.getHttpServer())
      .post('/posts')
      .send({
        title: 'Post',
        link: 'https://www.google.com/',
        content: 'My content',
        tags: ['tag1', 'tag2'],
      })
      .expect(401)
      .expect((response) => {
        expect(response.body).toMatchSnapshot();
      });
  });

  it('POST /posts (Authenticated)', async () => {
    return supertest(app.getHttpServer())
      .post('/posts')
      .set('Authorization', bearerToken)
      .send({
        title: 'Post',
        link: 'https://www.google.com/',
        content: 'My content',
        tags: ['tag1', 'tag2'],
      })
      .expect(201)
      .expect((response) => {
        expect(response.body).toMatchSnapshot({
          id: expect.any(Number),
          createdAt: expect.any(String),
          updatedAt: expect.any(String),
        });
      });
  });

  it('DELETE /posts/1012', async () => {
    return supertest(app.getHttpServer())
      .delete('/posts/1012')
      .expect(401)
      .expect((response) => {
        expect(response.body).toMatchSnapshot();
      });
  });

  it('DELETE /posts/1011 (Invalid user)', async () => {
    return supertest(app.getHttpServer())
      .delete('/posts/1011')
      .set('Authorization', bearerToken)
      .expect(403)
      .expect((response) => {
        expect(response.body).toMatchSnapshot();
      });
  });

  it('DELETE /posts/1013 (Authorized)', async () => {
    return supertest(app.getHttpServer()).delete('/posts/1013').set('Authorization', bearerToken).expect(200);
  });

  afterAll(async () => {
    await app.close();
  });
});
