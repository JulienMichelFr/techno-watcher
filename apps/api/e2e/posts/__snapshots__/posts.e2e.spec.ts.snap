// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Create comment Create a comment on a comment (Authenticated) 1`] = `
Object {
  "author": Object {
    "username": "User 1",
  },
  "content": "My comment",
  "createdAt": Any<String>,
  "deletedAt": null,
  "id": Any<Number>,
  "parentCommentId": 1001,
  "updatedAt": Any<String>,
}
`;

exports[`Posts Create comment Create a comment on a comment (Unauthenticated) 1`] = `
Object {
  "error": "Unauthorized",
  "message": "Token is missing",
  "statusCode": 401,
}
`;

exports[`Posts Create comment Create a comment on a post (Authenticated) 1`] = `
Object {
  "author": Object {
    "username": "User 1",
  },
  "content": "My comment",
  "createdAt": Any<String>,
  "deletedAt": null,
  "id": Any<Number>,
  "parentCommentId": null,
  "updatedAt": Any<String>,
}
`;

exports[`Posts Create comment Create a comment on a post (Unauthenticated) 1`] = `
Object {
  "error": "Unauthorized",
  "message": "Token is missing",
  "statusCode": 401,
}
`;

exports[`Posts Create post Create a post (Authenticated) 1`] = `
Object {
  "author": Object {
    "id": 1000,
    "username": "User 1",
  },
  "content": "My content",
  "createdAt": Any<String>,
  "id": Any<Number>,
  "link": "https://www.google.com/",
  "tags": Array [
    "tag1",
    "tag2",
  ],
  "title": "Post",
  "totalComments": 0,
  "updatedAt": Any<String>,
}
`;

exports[`Posts Create post Create a post (Unauthenticated) 1`] = `
Object {
  "error": "Unauthorized",
  "message": "Token is missing",
  "statusCode": 401,
}
`;

exports[`Posts Delete comment Delete a comment (Invalid user) 1`] = `
Object {
  "error": "Forbidden",
  "message": "You are not the author of this comment",
  "statusCode": 403,
}
`;

exports[`Posts Delete comment Delete a comment (Unauthenticated) 1`] = `
Object {
  "error": "Unauthorized",
  "message": "Token is missing",
  "statusCode": 401,
}
`;

exports[`Posts Delete post Delete a post (Invalid user) 1`] = `
Object {
  "error": "Forbidden",
  "message": "You are not allowed to delete this post",
  "statusCode": 403,
}
`;

exports[`Posts Delete post Delete a post (Unauthenticated) 1`] = `
Object {
  "error": "Unauthorized",
  "message": "Token is missing",
  "statusCode": 401,
}
`;

exports[`Posts Get comments on post Get comments on post with id 1013 1`] = `
Array [
  Object {
    "author": Object {
      "username": "User 1",
    },
    "content": "Minima soluta sed sed et optio explicabo at distinctio repudiandae. Magnam deleniti a ea. Non velit accusamus veniam qui. Necessitatibus velit ad aut officiis in officiis quasi. Sunt nulla dolores voluptatem esse magnam ut.",
    "createdAt": "2022-01-02T09:26:09.611Z",
    "deletedAt": null,
    "id": 1007,
    "parentCommentId": null,
    "updatedAt": "2022-01-02T09:26:09.611Z",
  },
  Object {
    "author": Object {
      "username": "User 2",
    },
    "content": "Est iste totam accusamus dolorem est. Quis non sit impedit similique incidunt odio aspernatur aut rem. Architecto est eum.",
    "createdAt": "2022-11-01T19:33:46.688Z",
    "deletedAt": null,
    "id": 1021,
    "parentCommentId": 1007,
    "updatedAt": "2022-11-01T19:33:46.688Z",
  },
]
`;

exports[`Posts Get posts Get a single post 1`] = `
Object {
  "author": Object {
    "id": 1000,
    "username": "User 1",
  },
  "content": "Et id harum unde et ratione minima non. Suscipit ipsum rem. Sed asperiores quaerat dolorum autem nihil voluptatem et hic ut. Molestiae rerum autem. Dolores nam soluta officia pariatur debitis.",
  "createdAt": "2022-12-21T13:50:49.826Z",
  "id": 1013,
  "link": "https://occaecati.name",
  "tags": Array [
    "enim",
    "laborum",
  ],
  "title": "Versatile national time-frame",
  "totalComments": 2,
  "updatedAt": "2022-12-21T13:50:49.826Z",
}
`;

exports[`Posts Get posts Get multiple posts 1`] = `
Object {
  "data": Array [
    Object {
      "author": Object {
        "id": 1000,
        "username": "User 1",
      },
      "content": "Et id harum unde et ratione minima non. Suscipit ipsum rem. Sed asperiores quaerat dolorum autem nihil voluptatem et hic ut. Molestiae rerum autem. Dolores nam soluta officia pariatur debitis.",
      "createdAt": "2022-12-21T13:50:49.826Z",
      "id": 1013,
      "link": "https://occaecati.name",
      "tags": Array [
        "enim",
        "laborum",
      ],
      "title": "Versatile national time-frame",
      "totalComments": 2,
      "updatedAt": "2022-12-21T13:50:49.826Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Similique molestiae id officia corporis quidem. Aliquam et ut eos ut nemo est voluptatem. Possimus ut quo labore. Alias amet quia enim. Quia ipsum pariatur facere illum esse recusandae veniam. Nihil enim fugit porro nam et quis sunt.",
      "createdAt": "2022-11-12T04:48:34.913Z",
      "id": 1002,
      "link": "https://quos.info",
      "tags": Array [
        "commodi",
        "aliquid",
      ],
      "title": "User-centric optimizing implementation",
      "totalComments": 1,
      "updatedAt": "2022-11-12T04:48:34.913Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Praesentium consequatur ut sit vel. Molestias fugiat quis cupiditate ipsa eos fugit est ullam. Sit labore et natus dolores ut quis eaque cupiditate. Et ut et et autem assumenda animi autem. Pariatur amet consequatur necessitatibus consequatur consequatur et explicabo sint. Nam sit dolore.",
      "createdAt": "2022-10-05T19:14:48.001Z",
      "id": 1003,
      "link": "https://consequuntur.com",
      "tags": Array [
        "aut",
        "ducimus",
      ],
      "title": "Ameliorated multimedia groupware",
      "totalComments": 2,
      "updatedAt": "2022-10-05T19:14:48.001Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Dolor officia a fuga omnis sit. Ut atque est nostrum. Quos aut quo eos vel velit autem et aspernatur.",
      "createdAt": "2022-07-15T20:50:58.831Z",
      "id": 1010,
      "link": "http://consequatur.dev",
      "tags": Array [
        "nostrum",
        "id",
      ],
      "title": "Realigned empowering monitoring",
      "totalComments": 3,
      "updatedAt": "2022-07-15T20:50:58.831Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Est quo facilis voluptas aperiam. Natus dolores quas ratione enim repellendus. Illum dolor repellendus voluptas.",
      "createdAt": "2022-07-11T05:07:16.307Z",
      "id": 1001,
      "link": "https://facilis.net",
      "tags": Array [
        "nihil",
        "sunt",
      ],
      "title": "Secured methodical frame",
      "totalComments": 4,
      "updatedAt": "2022-07-11T05:07:16.307Z",
    },
    Object {
      "author": Object {
        "id": 1000,
        "username": "User 1",
      },
      "content": "Minima soluta sed sed et optio explicabo at distinctio repudiandae. Magnam deleniti a ea. Non velit accusamus veniam qui. Necessitatibus velit ad aut officiis in officiis quasi. Sunt nulla dolores voluptatem esse magnam ut.",
      "createdAt": "2022-06-20T08:18:03.776Z",
      "id": 1004,
      "link": "http://omnis.com",
      "tags": Array [],
      "title": "Visionary fresh-thinking instruction set",
      "totalComments": 0,
      "updatedAt": "2022-06-20T08:18:03.776Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Aperiam autem non et aut illum ut nihil laborum omnis. Vitae et ab et. Cupiditate et est delectus. Mollitia qui qui dolores reiciendis perferendis voluptates maiores. Omnis corporis cumque sequi sequi excepturi velit laborum nobis. Neque id maiores voluptatem est debitis.",
      "createdAt": "2022-06-19T18:49:00.047Z",
      "id": 1007,
      "link": "http://unde.io",
      "tags": Array [
        "quos",
        "est",
      ],
      "title": "Customizable client-driven encoding",
      "totalComments": 0,
      "updatedAt": "2022-06-19T18:49:00.047Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Quia sit architecto itaque voluptas aliquam. Rem veritatis qui quasi sint velit dolorem maxime voluptatem. Sit aut laboriosam aspernatur dicta non consequatur qui recusandae. Dignissimos voluptatem labore praesentium.",
      "createdAt": "2022-06-13T18:29:25.771Z",
      "id": 1012,
      "link": "https://fugiat.io",
      "tags": Array [
        "nulla",
      ],
      "title": "Synergized 4th generation success",
      "totalComments": 1,
      "updatedAt": "2022-06-13T18:29:25.771Z",
    },
    Object {
      "author": Object {
        "id": 1000,
        "username": "User 1",
      },
      "content": "Minima soluta sed sed et optio explicabo at distinctio repudiandae. Magnam deleniti a ea. Non velit accusamus veniam qui. Necessitatibus velit ad aut officiis in officiis quasi. Sunt nulla dolores voluptatem esse magnam ut.",
      "createdAt": "2022-06-06T13:34:27.804Z",
      "id": 1009,
      "link": "http://vitae.name",
      "tags": Array [
        "nemo",
        "fugiat",
      ],
      "title": "Enhanced responsive software",
      "totalComments": 1,
      "updatedAt": "2022-06-06T13:34:27.804Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Aut facilis qui. Cupiditate sit ratione eum sunt rerum impedit. Qui suscipit debitis et et voluptates voluptatem voluptatibus. Quas voluptatum quae corporis corporis possimus.",
      "createdAt": "2022-05-23T12:54:50.455Z",
      "id": 1006,
      "link": "http://qui.info",
      "tags": Array [
        "consectetur",
      ],
      "title": "Re-engineered neutral database",
      "totalComments": 0,
      "updatedAt": "2022-05-23T12:54:50.455Z",
    },
  ],
  "from": 0,
  "perPage": 10,
  "to": 10,
  "total": 15,
}
`;

exports[`Posts Get posts Get multiple posts with pagination 1`] = `
Object {
  "data": Array [
    Object {
      "author": Object {
        "id": 1000,
        "username": "User 1",
      },
      "content": "Quo voluptatem quia numquam laudantium sit quibusdam aut. Veritatis omnis neque ea saepe hic enim. Nam odit dolor non consequuntur perspiciatis inventore ut sint. Velit quod praesentium adipisci modi.",
      "createdAt": "2022-05-08T02:58:30.546Z",
      "id": 1011,
      "link": "https://omnis.name",
      "tags": Array [
        "ducimus",
      ],
      "title": "Configurable zero administration projection",
      "totalComments": 2,
      "updatedAt": "2022-05-08T02:58:30.546Z",
    },
    Object {
      "author": Object {
        "id": 1000,
        "username": "User 1",
      },
      "content": "Veniam commodi autem voluptatibus eos dolor quas reprehenderit. Praesentium cupiditate tempore et reprehenderit. Deleniti exercitationem illum maiores. Reprehenderit odio in ea voluptatem ut ut ullam.",
      "createdAt": "2022-05-03T11:16:10.308Z",
      "id": 1014,
      "link": "http://tenetur.name",
      "tags": Array [
        "beatae",
      ],
      "title": "Monitored bottom-line productivity",
      "totalComments": 0,
      "updatedAt": "2022-05-03T11:16:10.308Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Labore corporis blanditiis dolorum nemo nam praesentium alias sequi inventore. Cupiditate rerum enim sint quis. Eum occaecati provident labore veniam deserunt vero sed soluta repellat. Cum sapiente pariatur et ea a recusandae et optio. Sequi doloribus reiciendis corrupti quidem accusamus est nesciunt. Excepturi accusamus consequatur est sed maiores excepturi autem.",
      "createdAt": "2022-03-11T17:58:58.835Z",
      "id": 1015,
      "link": "http://sit.name",
      "tags": Array [
        "fugiat",
      ],
      "title": "Ameliorated asynchronous pricing structure",
      "totalComments": 6,
      "updatedAt": "2022-03-11T17:58:58.835Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Sunt velit facere fuga et voluptas inventore itaque. Necessitatibus ratione in esse. Quasi dignissimos quia est sequi incidunt enim reiciendis. At omnis iure in doloremque. Aut tempore consequatur facilis est ut distinctio est quas. Autem sunt est saepe quasi sed reprehenderit error magnam.",
      "createdAt": "2022-02-09T12:33:59.515Z",
      "id": 1008,
      "link": "https://sapiente.com",
      "tags": Array [
        "facilis",
        "nihil",
        "id",
      ],
      "title": "Focused logistical definition",
      "totalComments": 0,
      "updatedAt": "2022-02-09T12:33:59.515Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Vel et molestiae quis ea modi quas tempore dolorum fuga. Aut dolore numquam et. Amet sit quibusdam ea blanditiis consectetur velit.",
      "createdAt": "2022-02-04T23:23:55.111Z",
      "id": 1005,
      "link": "https://quas.net",
      "tags": Array [
        "sit",
      ],
      "title": "Distributed heuristic archive",
      "totalComments": 3,
      "updatedAt": "2022-02-04T23:23:55.111Z",
    },
  ],
  "from": 10,
  "perPage": 10,
  "to": 20,
  "total": 15,
}
`;

exports[`Posts Get posts Get multiple posts with sort 1`] = `
Object {
  "data": Array [
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Est quo facilis voluptas aperiam. Natus dolores quas ratione enim repellendus. Illum dolor repellendus voluptas.",
      "createdAt": "2022-07-11T05:07:16.307Z",
      "id": 1001,
      "link": "https://facilis.net",
      "tags": Array [
        "nihil",
        "sunt",
      ],
      "title": "Secured methodical frame",
      "totalComments": 4,
      "updatedAt": "2022-07-11T05:07:16.307Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Similique molestiae id officia corporis quidem. Aliquam et ut eos ut nemo est voluptatem. Possimus ut quo labore. Alias amet quia enim. Quia ipsum pariatur facere illum esse recusandae veniam. Nihil enim fugit porro nam et quis sunt.",
      "createdAt": "2022-11-12T04:48:34.913Z",
      "id": 1002,
      "link": "https://quos.info",
      "tags": Array [
        "commodi",
        "aliquid",
      ],
      "title": "User-centric optimizing implementation",
      "totalComments": 1,
      "updatedAt": "2022-11-12T04:48:34.913Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Praesentium consequatur ut sit vel. Molestias fugiat quis cupiditate ipsa eos fugit est ullam. Sit labore et natus dolores ut quis eaque cupiditate. Et ut et et autem assumenda animi autem. Pariatur amet consequatur necessitatibus consequatur consequatur et explicabo sint. Nam sit dolore.",
      "createdAt": "2022-10-05T19:14:48.001Z",
      "id": 1003,
      "link": "https://consequuntur.com",
      "tags": Array [
        "aut",
        "ducimus",
      ],
      "title": "Ameliorated multimedia groupware",
      "totalComments": 2,
      "updatedAt": "2022-10-05T19:14:48.001Z",
    },
    Object {
      "author": Object {
        "id": 1000,
        "username": "User 1",
      },
      "content": "Minima soluta sed sed et optio explicabo at distinctio repudiandae. Magnam deleniti a ea. Non velit accusamus veniam qui. Necessitatibus velit ad aut officiis in officiis quasi. Sunt nulla dolores voluptatem esse magnam ut.",
      "createdAt": "2022-06-20T08:18:03.776Z",
      "id": 1004,
      "link": "http://omnis.com",
      "tags": Array [],
      "title": "Visionary fresh-thinking instruction set",
      "totalComments": 0,
      "updatedAt": "2022-06-20T08:18:03.776Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Vel et molestiae quis ea modi quas tempore dolorum fuga. Aut dolore numquam et. Amet sit quibusdam ea blanditiis consectetur velit.",
      "createdAt": "2022-02-04T23:23:55.111Z",
      "id": 1005,
      "link": "https://quas.net",
      "tags": Array [
        "sit",
      ],
      "title": "Distributed heuristic archive",
      "totalComments": 3,
      "updatedAt": "2022-02-04T23:23:55.111Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Aut facilis qui. Cupiditate sit ratione eum sunt rerum impedit. Qui suscipit debitis et et voluptates voluptatem voluptatibus. Quas voluptatum quae corporis corporis possimus.",
      "createdAt": "2022-05-23T12:54:50.455Z",
      "id": 1006,
      "link": "http://qui.info",
      "tags": Array [
        "consectetur",
      ],
      "title": "Re-engineered neutral database",
      "totalComments": 0,
      "updatedAt": "2022-05-23T12:54:50.455Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Aperiam autem non et aut illum ut nihil laborum omnis. Vitae et ab et. Cupiditate et est delectus. Mollitia qui qui dolores reiciendis perferendis voluptates maiores. Omnis corporis cumque sequi sequi excepturi velit laborum nobis. Neque id maiores voluptatem est debitis.",
      "createdAt": "2022-06-19T18:49:00.047Z",
      "id": 1007,
      "link": "http://unde.io",
      "tags": Array [
        "quos",
        "est",
      ],
      "title": "Customizable client-driven encoding",
      "totalComments": 0,
      "updatedAt": "2022-06-19T18:49:00.047Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Sunt velit facere fuga et voluptas inventore itaque. Necessitatibus ratione in esse. Quasi dignissimos quia est sequi incidunt enim reiciendis. At omnis iure in doloremque. Aut tempore consequatur facilis est ut distinctio est quas. Autem sunt est saepe quasi sed reprehenderit error magnam.",
      "createdAt": "2022-02-09T12:33:59.515Z",
      "id": 1008,
      "link": "https://sapiente.com",
      "tags": Array [
        "facilis",
        "nihil",
        "id",
      ],
      "title": "Focused logistical definition",
      "totalComments": 0,
      "updatedAt": "2022-02-09T12:33:59.515Z",
    },
    Object {
      "author": Object {
        "id": 1000,
        "username": "User 1",
      },
      "content": "Minima soluta sed sed et optio explicabo at distinctio repudiandae. Magnam deleniti a ea. Non velit accusamus veniam qui. Necessitatibus velit ad aut officiis in officiis quasi. Sunt nulla dolores voluptatem esse magnam ut.",
      "createdAt": "2022-06-06T13:34:27.804Z",
      "id": 1009,
      "link": "http://vitae.name",
      "tags": Array [
        "nemo",
        "fugiat",
      ],
      "title": "Enhanced responsive software",
      "totalComments": 1,
      "updatedAt": "2022-06-06T13:34:27.804Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Dolor officia a fuga omnis sit. Ut atque est nostrum. Quos aut quo eos vel velit autem et aspernatur.",
      "createdAt": "2022-07-15T20:50:58.831Z",
      "id": 1010,
      "link": "http://consequatur.dev",
      "tags": Array [
        "nostrum",
        "id",
      ],
      "title": "Realigned empowering monitoring",
      "totalComments": 3,
      "updatedAt": "2022-07-15T20:50:58.831Z",
    },
  ],
  "from": 0,
  "perPage": 10,
  "to": 10,
  "total": 15,
}
`;

exports[`Posts Get posts Get multiple posts with tags 1`] = `
Object {
  "data": Array [
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Est quo facilis voluptas aperiam. Natus dolores quas ratione enim repellendus. Illum dolor repellendus voluptas.",
      "createdAt": "2022-07-11T05:07:16.307Z",
      "id": 1001,
      "link": "https://facilis.net",
      "tags": Array [
        "nihil",
        "sunt",
      ],
      "title": "Secured methodical frame",
      "totalComments": 4,
      "updatedAt": "2022-07-11T05:07:16.307Z",
    },
    Object {
      "author": Object {
        "id": 1000,
        "username": "User 1",
      },
      "content": "Minima soluta sed sed et optio explicabo at distinctio repudiandae. Magnam deleniti a ea. Non velit accusamus veniam qui. Necessitatibus velit ad aut officiis in officiis quasi. Sunt nulla dolores voluptatem esse magnam ut.",
      "createdAt": "2022-06-06T13:34:27.804Z",
      "id": 1009,
      "link": "http://vitae.name",
      "tags": Array [
        "nemo",
        "fugiat",
      ],
      "title": "Enhanced responsive software",
      "totalComments": 1,
      "updatedAt": "2022-06-06T13:34:27.804Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Labore corporis blanditiis dolorum nemo nam praesentium alias sequi inventore. Cupiditate rerum enim sint quis. Eum occaecati provident labore veniam deserunt vero sed soluta repellat. Cum sapiente pariatur et ea a recusandae et optio. Sequi doloribus reiciendis corrupti quidem accusamus est nesciunt. Excepturi accusamus consequatur est sed maiores excepturi autem.",
      "createdAt": "2022-03-11T17:58:58.835Z",
      "id": 1015,
      "link": "http://sit.name",
      "tags": Array [
        "fugiat",
      ],
      "title": "Ameliorated asynchronous pricing structure",
      "totalComments": 6,
      "updatedAt": "2022-03-11T17:58:58.835Z",
    },
    Object {
      "author": Object {
        "id": 1001,
        "username": "User 2",
      },
      "content": "Sunt velit facere fuga et voluptas inventore itaque. Necessitatibus ratione in esse. Quasi dignissimos quia est sequi incidunt enim reiciendis. At omnis iure in doloremque. Aut tempore consequatur facilis est ut distinctio est quas. Autem sunt est saepe quasi sed reprehenderit error magnam.",
      "createdAt": "2022-02-09T12:33:59.515Z",
      "id": 1008,
      "link": "https://sapiente.com",
      "tags": Array [
        "facilis",
        "nihil",
        "id",
      ],
      "title": "Focused logistical definition",
      "totalComments": 0,
      "updatedAt": "2022-02-09T12:33:59.515Z",
    },
  ],
  "from": 0,
  "perPage": 10,
  "to": 10,
  "total": 4,
}
`;
